src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -118,6 +130,13 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         quiet.set(false)
         pull.set(false)
         cacheFrom.empty()
+        String safeTaskPath = path.replaceFirst("^:", "").replaceAll(":", "_")
+        imageIdFile.set(new File(project.buildDir, ".docker/${safeTaskPath}-imageId.txt"))
+
+        if(imageIdFile.get().asFile.exists()){
+            // Get all images in docker registry and make sure the imageId is found in there
+            outputs.upToDateWhen{getDockerClient().listImagesCmd().exec().any{image -> image.toString().contains(imageIdFile.get().asFile.text)}}
```

Comment:
Can you please make the Closure body a multi-line statement? At the moment it's not very readable.